lib/fluent/plugin/buffer.rb

```diff
@@ -60,7 +60,17 @@ class BufferChunkOverflowError < BufferError; end # A record size is larger than
       desc 'Compress buffered data.'
       config_param :compress, :enum, list: [:text, :gzip], default: :text
 
-      Metadata = Struct.new(:timekey, :tag, :variables) do
+      Metadata = Struct.new(:timekey, :tag, :variables, :seq) do
```

Comment:
No need to use `seq` in `cmp_variables` and `<=>`?

Reply:
Yes. `seq` is not related to an order. it's just used for the uniqueness of metadata.